db/explorerParser.ts

```diff
@@ -0,0 +1,132 @@
+// This code was adapted from https://github.com/owid/automation/blob/main/automation/mirror_explorers.py
+
+import { ExplorerGrammar } from "@ourworldindata/explorer"
+
+interface Statement {
+    verb: string
+    args: string[]
+    block?: Record<string, any>[]
+}
+
+const JSON_VERSION = 1
+
+// Constant (non-array) fields
+const FLAGS: Set<string> = new Set(
+    Object.keys(ExplorerGrammar).filter(
+        (key) =>
+            ![
+                "table",
+                "graphers",
+                "columns",
+                "selection",
+                "pickerColumnSlugs",
+                "yVariableIds",
+            ].includes(key)
+    )
+)
+
+function isBlock(verb: string, args: string[]): boolean {
+    if (verb === "graphers" || verb === "columns") return true
+    return verb === "table" && !args.length
+}
+
+function readBlock(
+    lines: string[],
+    slug: string,
+    start: number
+): [Record<string, any>[], number] {
+    const block: string[] = []
+    let i = start
+    while (i < lines.length && lines[i].startsWith("\t")) {
+        block.push(lines[i].slice(1).replace(/\r?$/, ""))
+        i++
+    }
+    const records = tsvToRecords(block, slug, start)
+    pruneNulls(records)
+    return [records, i - start]
+}
+
+function pruneNulls(records: Record<string, any>[]): void {
+    for (const r of records) {
+        for (const k of Object.keys(r)) {
+            if (!r[k]) delete r[k]
+        }
+    }
+}
+
+function tsvToRecords(
+    block: string[],
+    slug: string,
+    start: number
+): Record<string, any>[] {
```

Comment:
I think we can turn all `Record<string, any>` in this file into `Record<string, string>`.